lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
@anupam128 ASA has done cursory investigation into Windows Job Object - A few things to consider:

1. I'm under the impression that Windows Job Object not supported on HDInsight by default.
2. In the ASA case, it's not necessary that there will be perf improvements if Job Objects are enabled, provided that YARN checks for the memory at a lower frequency than our thread running GC so we have enough control.
3. Also, with Job Object enabled, as the process approaches the memory limit, blocking GC will be triggered much more often and sometimes it's better to restart the entire process instead.

The bottom line is that more measurements are required as to how enabling Job Objects affects performance. I wasn't assigned the task to look into Job Objects so my understanding here may be shallow or misguided.